src/cuga/backend/cuga_graph/nodes/cuga_lite/executors/tests/test_local_sandbox_skills.py

```
@@ -0,0 +1,379 @@
+    # Relative shell output landed in the per-thread workspace
+    out_file = expected / "out.txt"
+    assert out_file.exists(), f"out.txt should be under {expected}; stdout={stdout!r} stderr={stderr!r}"
+    assert out_file.read_text().strip() == "hello"
+
+    # Parent-level file MUST NOT auto-propagate into the per-thread dir.
+    assert not (expected / "sibling.txt").exists(), (
+        "per-thread workspaces must stay isolated — no auto-copy from parent"
+    )
+
+    # Nothing leaked into /tmp/cuga (the legacy location)
+    assert not (Path("/tmp") / "cuga").exists() or not any(Path("/tmp/cuga").iterdir()), (
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp"

(S108)